var/spack/repos/builtin/packages/py-datalad/package.py

```diff
@@ -73,8 +83,7 @@ class PyDatalad(PythonPackage):
     depends_on("py-jsmin", when="@:0.14", type=("build", "run"))
 
     # metadata
-    depends_on("py-simplejson", type=("build", "run"))
-    depends_on("py-whoosh", type=("build", "run"))
+    conflicts("~metadata", when="@:0.17")
```

Comment:
This line isn't needed, the `when` in the variant definition will handle this

Reply:
For older version the metadata dependencies are needed, but for the current version not anymore. I changed that now.